File: backends/ruby-hanami/app/infrastructure/model/notification.rb
# frozen_string_literal: true
#
# 通知モデルを定義するモジュール。
#
# @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

module Infrastructure
  module Model
    # 通知の永続化モデルです。
    # @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
    Notification = Struct.new(
      :id, :staff_id, :message_type, :title, :message, :url, :read,
      :created_at, :created_by, :updated_at, :updated_by,
      :deleted_at, :deleted_by, :version,
      keyword_init: true
    )
  end
end

File: backends/go-echo/ent/schema/notification.go
package schema

import (
	"entgo.io/ent"
	"entgo.io/ent/dialect/entsql"
	"entgo.io/ent/schema"
	"entgo.io/ent/schema/field"
	"time"
)

// Notification holds the schema definition for the Notification entity.
type Notification struct {
	ent.Schema
}

func (Notification) Annotations() []schema.Annotation {
	return []schema.Annotation{
		entsql.Annotation{Table: "notifications"},
	}
}

func (Notification) Fields() []ent.Field {
	return []ent.Field{
		field.Uint64("id").Positive().Immutable(),
		field.Uint("staff_id"),
		field.Int("message_type"),
		field.String("title"),
		field.String("message"),
		field.String("url").Optional().Nillable(),
		field.Bool("read").Default(false),
		field.Time("created_at").Default(time.Now),
		field.Uint("created_by").Optional().Nillable(),
		field.Time("updated_at").Default(time.Now),
		field.Uint("updated_by").Optional().Nillable(),
		field.Time("deleted_at").Optional().Nillable(),
		field.Uint("deleted_by").Optional().Nillable(),
		field.Int("version").Default(1),
	}
}

func (Notification) Edges() []ent.Edge {
	return nil
}
